core/src/test/resources/org/apache/calcite/test/SqlToRelConverterTest.xml

```diff
@@ -4170,14 +4170,61 @@ LogicalTableModify(table=[[CATALOG, SALES, EMPNULLABLES_20]], operation=[MERGE],
       <![CDATA[
 LogicalTableModify(table=[[CATALOG, SALES, EMPNULLABLES_20]], operation=[MERGE], updateColumnList=[[]], flattened=[true])
-  LogicalProject($f0=[=($14, $5)], EMPNO=[$0], ENAME=[CAST($1):VARCHAR(20)], $f3=[null:VARCHAR(10)], $f4=[null:INTEGER], $f5=[null:TIMESTAMP(0)], SAL=[CAST($5):INTEGER], $f7=[null:INTEGER], $f8=[20], $f9=[null:BOOLEAN])
+  LogicalProject(EMPNO=[$0], ENAME=[CAST($1):VARCHAR(20)], $f2=[null:VARCHAR(10)], $f3=[null:INTEGER], $f4=[null:TIMESTAMP(0)], SAL=[CAST($5):INTEGER], $f6=[null:INTEGER], $f7=[20], $f8=[null:BOOLEAN])
     LogicalJoin(condition=[=($14, $5)], joinType=[left])
       LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7], SLACKER=[$8])
         LogicalFilter(condition=[=($7, 30)])
           LogicalTableScan(table=[[CATALOG, SALES, EMP]])
       LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], SLACKER=[$8])
         LogicalFilter(condition=[AND(=($7, 20), >($5, 1000))])
           LogicalTableScan(table=[[CATALOG, SALES, EMPNULLABLES]])
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testMergeIntoDelete">
+    <Resource name="sql">
+      <![CDATA[merge into empnullables_20 as target
+        using (select * from emp where deptno = 30) as source
+        on target.empno = source.empno
+        when matched then
+          delete
+        when not matched then
+          insert (empno, sal, ename)
+          values (ABS(source.empno), source.sal, source.ename)]]>
+    </Resource>
+    <Resource name="plan">
+      <![CDATA[
+LogicalTableModify(table=[[CATALOG, SALES, EMPNULLABLES_20]], operation=[MERGE], updateColumnList=[[]], flattened=[true])
+  LogicalProject($f0=[ABS($0)], ENAME=[CAST($1):VARCHAR(20)], $f2=[null:VARCHAR(10)], $f3=[null:INTEGER], $f4=[null:TIMESTAMP(0)], SAL=[CAST($5):INTEGER], $f6=[null:INTEGER], $f7=[20], $f8=[null:BOOLEAN])
+    LogicalJoin(condition=[=($9, $0)], joinType=[left])
+      LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], DEPTNO=[$7], SLACKER=[$8])
+        LogicalFilter(condition=[=($7, 30)])
+          LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+      LogicalProject(EMPNO=[$0], ENAME=[$1], JOB=[$2], MGR=[$3], HIREDATE=[$4], SAL=[$5], COMM=[$6], SLACKER=[$8])
+        LogicalFilter(condition=[AND(=($7, 20), >($5, 1000))])
+          LogicalTableScan(table=[[CATALOG, SALES, EMPNULLABLES]])
+]]>
+    </Resource>
+  </TestCase>
+  <TestCase name="testMergeIntoDeleteOnly">
+    <Resource name="sql">
+      <![CDATA[merge into empnullables_20 as target
+        using (select * from emp where deptno = 30) as source
+        on target.empno = source.empno
+        when matched then
+          delete]]>
+    </Resource>
+    <Resource name="plan">
+      <![CDATA[
+LogicalTableModify(table=[[CATALOG, SALES, EMPNULLABLES_20]], operation=[MERGE], updateColumnList=[[]], flattened=[true])
+  LogicalProject
```

Comment:
This is clearly not correct for what we need, but I figured it's acceptable since we need to universally change these projections everywhere.

Reply:
I agree that this is ok for now, and we can revisit this later